### PR TITLE
resolved: two more tweaks to the stub

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/resolve/resolved-dns-stub.c [hunk 1]
```diff
@@ -771,21 +722,63 @@ static void dns_stub_query_complete(DnsQuery *q) {
...
+                                dns_query_free(q);
+                                return;
+                        }
+
+                        if (cname_result == DNS_QUERY_MATCH) /* A match? Then we are done, let's return what we got */
+                                break;
+
+                        /* We followed a CNAME. and collected the RRs that answer the redirected question
+                         * successfully. Let's not try to do this again. */
+                        assert(cname_result = DNS_QUERY_CNAME);
```
Comment (yuwata) [inline src/resolve/resolved-dns-stub.c]: `=` -> `==`

**Comment by yuwata** [comment]
Maybe, it is better to split the second commit into two? First split `dns_query_process_cname()` into `_one()` and `_many()`, then fix stub?

**Comment by yuwata** [comment]
Are there any linked issue or reproducers?

File: src/resolve/resolved-dns-stub.c (hunk 1 above)
Comment (poettering) [reply src/resolve/resolved-dns-stub.c]: ouch ouch!

(but i guess ultimately doesn't matter)

**Comment by poettering** [comment]
> Are there any linked issue or reproducers?

This was mostly discussed with @keszybz on telegram.

**Comment by poettering** [comment]
> Maybe, it is better to split the second commit into two? First split `dns_query_process_cname()` into `_one()` and `_many()`, then fix stub?

Done! PTAL!

**Comment by poettering** [comment]
Yay, all green!

**Comment by keszybz** [comment]
LGTM.

**Comment by keszybz** [comment]
This definitely fixes the assignment of answer to sections, the second commit.
The third commit is harder to verify. It seems to work all fine, but I don't know how to reproduce the original issue.

**Pull request merged**